src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -168,9 +175,87 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
     }
   }
 
+  initUserConfig () {
+    if (this.autoSave) {
+      this.tableName = this._calculateUniqueTableName(this.columns);
+      this._loadUserConfig();
+      this._initUserConfigAutoSave();
+    }
+    if (!this.userConfig.limit) {
+      this.userConfig.limit = this.limit;
+    }
+    if (!this.userConfig.sorts) {
+      this.userConfig.sorts = this.sorts;
+    }
+    if (!this.userConfig.columns) {
+      this.updateUserColumns();
+    } else {
+      this.columns.forEach((c, i) => {
+        c.isHidden = this.userConfig.columns[i].isHidden;
+      });
+    }
+  }
+
+  _calculateUniqueTableName (columns) {
+    const stringToNumber = (s) => {
+      if (!_.isString(s)) {
+        return 0;
+      }
+      let result = 0;
+      for (let i = 0; i < s.length; i++) {
+        result += s.charCodeAt(i) * i;
+      }
+      return result;
+    };
+    return columns.reduce((result, value, index) =>
+      ((stringToNumber(value.prop) + stringToNumber(value.name)) * (index + 1)) + result,
+      0).toString();
+  }
+
+  _loadUserConfig () {
+    const loaded = this.localStorage.getItem(this.tableName);
+    if (loaded) {
+      this.userConfig = JSON.parse(loaded);
+    }
+  }
+
+  _initUserConfigAutoSave() {
+    const source = Observable.create(this._initUserConfigProxy.bind(this));
+    this.saveSubscriber = source.subscribe(this._saveUserConfig.bind(this));
+  }
+
+  _initUserConfigProxy (observer) {
+    this.userConfig = new Proxy(this.userConfig, {
+      set(config, prop, value) {
+        config[prop] = value;
+        observer.next(config);
+        return true;
+      }
+    });
+  }
+
+  _saveUserConfig (config) {
+    this.localStorage[this.tableName] = JSON.stringify(config);
```

Comment:
Please replace this with `this.localStorage.setItem(this.tableName, JSON.stringify(config))`. 
This change is needed for Jest.